crates/goose-cli/src/commands/web.rs

```diff
@@ -158,12 +158,17 @@ pub async fn handle_web(
 
     let model_config = goose::model::ModelConfig::new(&model)?;
 
-    // Create the agent
+    let init_session = SessionManager::create_session(
+        std::env::current_dir().unwrap_or_else(|_| std::path::PathBuf::from(".")),
+        "Web Agent Initialization".to_string(),
+        SessionType::Hidden,
+    )
+    .await?;
+
     let agent = Agent::new();
     let provider = goose::providers::create(&provider_name, model_config).await?;
-    agent.update_provider(provider).await?;
+    agent.update_provider(provider, &init_session.id).await?;
```

Comment:
I think we should move this to where the session is created, not create a specific session for it and also update the agent there

then if you resume a session by going to /<session_id> we can in theory load the provider, model back from the session, which is ultimately the goal of this whole exercise